### PR TITLE
feat: add university and seasonal alerts

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

The Lean alert section will display an alert for each of the following conditions:
1) if the research recruitment flag is on.
2) if the university creation flag is on and the user is not donating.
3) if the seasonal message flag is on.

The feature flags are controlled through our feature flag platfom (growthbook)
Currently only the university creation flag is on.

### CODE REVIEW

File: client/src/components/Intro/learn-alert.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+const LearnAlert = ({
+  onDonationAlertClick,
+  isDonating
+}: LearnAlertProps): JSX.Element | null => {
+  const { t } = useTranslation();
+  const researchRecruitment = useFeature('show-research-recruitment-alert');
+  const universityCreation = useFeature('university-creation-alert');
+  const seasionalMessage = useFeature('seasonal-alert');
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/learn-alert.tsx:19]:
```suggestion
  const seasonalMessage = useFeature('seasonal-alert');
```
Typo

File: client/src/components/Intro/learn-alert.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        <Button
+          href={'https://wharton.qualtrics.com/jfe/form/SV_57rJfXROkQDDU2y'}
+        >
+          Learn about HabitLab
+        </Button>
+      </div>
+    </Alert>
+  );
+
+  const seasionalMessageAlert = (
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/learn-alert.tsx:42]:
```suggestion
  const seasonalMessageAlert = (
```

File: client/src/components/Intro/learn-alert.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          {t('donate.become-supporter')}
+        </Link>
+      </p>
+    </Alert>
+  );
+
+  if (researchRecruitment.on) return researchRecruitmentAlert;
+  else if (universityCreation.on && !isDonating) return universityCreationAlert;
+  else if (seasionalMessage.on) return seasionalMessageAlert;
+  else return null;
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/learn-alert.tsx:87]:
```suggestion
  if (researchRecruitment.on) return researchRecruitmentAlert;
  if (universityCreation.on && !isDonating) return universityCreationAlert;
  if (seasonalMessage.on) return seasonalMessageAlert;
  return null;
```
Nitpick, really, but if you're returning early the elses are redundant. (also I fixed the typos!)

File: client/src/components/Intro/index.tsx [hunk 4]
```diff
@@ -16,6 +16,8 @@ interface IntroProps {
   pending?: boolean;
   slug?: string;
   username?: string;
+  onDonationAlertClick?: () => void;
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/index.tsx:19]:
```suggestion
  onDonationAlertClick: () => void;
```
It's always present.

File: client/src/components/Intro/learn-alert.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { Alert, Button } from '@freecodecamp/react-bootstrap';
+import { useFeature } from '@growthbook/growthbook-react';
+import { useTranslation } from 'react-i18next';
+import { Link } from '../helpers';
+
+interface LearnAlertProps {
+  onDonationAlertClick?: () => void;
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/learn-alert.tsx:8]:
```suggestion
  onDonationAlertClick: () => void;
```

File: client/src/components/Intro/intro.test.tsx [hunk 6]
```diff
@@ -60,7 +60,8 @@ const loggedInProps = {
   navigate: () => jest.fn(),
   pending: false,
   slug: '/',
-  username: 'DevelopmentUser'
+  username: 'DevelopmentUser',
+  isDonating: false
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/intro.test.tsx:64]:
```suggestion
  isDonating: false,
  onDonationAlertClick: () => jest.fn()
```

File: client/src/components/Intro/intro.test.tsx [hunk 7]
```diff
@@ -70,5 +71,6 @@ const loggedOutProps = {
   navigate: () => jest.fn(),
   pending: false,
   slug: '/',
-  username: ''
+  username: '',
+  isDonating: false
```
Comment (ojeytonwilliams) [inline client/src/components/Intro/intro.test.tsx:75]:
```suggestion
  isDonating: false,
  onDonationAlertClick: () => jest.fn()
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Looks nice @ahmadabdolsaheb.  All I found were minor nitpicks:

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
LGTM :+1:

**Review by ojeytonwilliams — APPROVED** [review]

File: client/i18n/locales/english/translations.json [hunk 8]
```diff
@@ -325,7 +325,11 @@
     "sorry-getting-there": "Sorry, your code does not pass. You're getting there.",
     "sorry-hang-in-there": "Sorry, your code does not pass. Hang in there.",
     "sorry-dont-giveup": "Sorry, your code does not pass. Don't give up.",
-    "challenges-completed": "{{completedCount}} of {{totalChallenges}} challenges completed"
+    "challenges-completed": "{{completedCount}} of {{totalChallenges}} challenges completed",
+    "season-greetings-fcc": "Season's Greetings from the freeCodeCamp community 🎉",
+    "if-getting-value": "If you're getting a lot out of freeCodeCamp, now is a great time to donate to support our nonprofit's mission.",
+    "building-a-university": "We're Building a Free Computer Science University Degree Program",
+    "if-help-univeristy": "We've already made a ton of progress. Support our charity with the long road ahead."
```
Comment (ShaunSHamilton) [inline client/i18n/locales/english/translations.json:332]: _"with"_ sounded weird.

```suggestion
    "if-help-univeristy": "We've already made a ton of progress. Support our charity for the long road ahead."
```

File: client/src/components/Intro/learn-alert.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+      </p>
+    </Alert>
+  );
+
+  const universityCreationAlert = (
+    <Alert bsStyle='info' className='annual-donation-alert'>
+      <p>
+        <b>{t('learn.building-a-university')}</b>
+      </p>
+      <p>{t('learn.if-help-univeristy')}</p>
```
Comment (ShaunSHamilton) [inline client/src/components/Intro/learn-alert.tsx:68]:
```suggestion
      <p>{t('learn.if-help-university')}</p>
```

File: client/i18n/locales/english/translations.json (hunk 8 above)
Comment (ahmadabdolsaheb) [reply client/i18n/locales/english/translations.json:332]: Quincy's copy. Should I double check with him?

Comment (ShaunSHamilton) [reply client/i18n/locales/english/translations.json:332]: That, or edit the suggestion to just fix the `university` typo

Comment (ahmadabdolsaheb) [inline client/i18n/locales/english/translations.json:332]:
```suggestion
    "if-help-university": "We've already made a ton of progress. Support our charity with the long road ahead."
```

**Review by Sembauke — APPROVED** [review]

File: client/src/components/Intro/learn-alert.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      <p style={{ marginBottom: 20, marginTop: 14 }}>
+        Would you like to get involved? You’ll get free coaching from our
+        scientists.
+      </p>
+      <div style={{ display: 'flex', justifyContent: 'center' }}>
+        <Button
+          href={'https://wharton.qualtrics.com/jfe/form/SV_57rJfXROkQDDU2y'}
+        >
+          Learn about HabitLab
+        </Button>
```
Comment (ShaunSHamilton) [inline client/src/components/Intro/learn-alert.tsx:37]: I do not like being taken away from the main site. Can we make this an `a[target="_blank"]`?

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Other than this, this LGTM 🛑

**Review by ShaunSHamilton — APPROVED** [review]
LBTM 👍

**Comment by ahmadabdolsaheb** [comment]
LBTM 🤔

**Pull request merged**